pkg/storage/etcd/etcd_helper.go

```diff
@@ -42,64 +44,107 @@ import (
 )
 
 // storage.Config object for etcd.
-type EtcdConfig struct {
-	ServerList []string
-	Codec      runtime.Codec
-	Prefix     string
-	Quorum     bool
+type EtcdStorageConfig struct {
+	Config EtcdConfig
+	Codec  runtime.Codec
 }
 
 // implements storage.Config
-func (c *EtcdConfig) GetType() string {
+func (c *EtcdStorageConfig) GetType() string {
 	return "etcd"
 }
 
 // implements storage.Config
-func (c *EtcdConfig) NewStorage() (storage.Interface, error) {
-	cfg := etcd.Config{
-		Endpoints: c.ServerList,
-		// TODO: Determine if transport needs optimization
-		Transport: &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			Dial: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-			}).Dial,
-			TLSHandshakeTimeout: 10 * time.Second,
-			MaxIdleConnsPerHost: 500,
-		},
-	}
-	etcdClient, err := etcd.New(cfg)
+func (c *EtcdStorageConfig) NewStorage() (storage.Interface, error) {
+	etcdClient, err := etcd.New(c.Config.getEtcdClientConfig())
 	if err != nil {
 		return nil, err
 	}
-	return NewEtcdStorage(etcdClient, c.Codec, c.Prefix, c.Quorum), nil
+	return NewEtcdStorage(etcdClient, c.Codec, c.Config.Prefix, c.Config.Quorum), nil
+}
+
+// Configuration object for constructing etcd.Client
+// and accessing etcd.
+type EtcdConfig struct {
+	ServerList []string
+	Prefix     string
```

Comment:
the line between EtcdConfig and EtcdStorageConfig is unclear... I would expect Prefix in EtcdStorageConfig